validation/uniqueness_test.go

```diff
@@ -20,23 +21,39 @@ type uniqueProperties struct {
 	RPC       []string
 }
 
-type chainIDSet map[uint64]bool
+type (
+	chainIDSet   map[uint64]bool
+	chainNameSet map[string]bool
+)
 
-func (s chainIDSet) AddIfUnique(id uint64) error {
-	if s[id] {
-		return fmt.Errorf("Chain with ID %d is duplicated", id)
-	}
-	s[id] = true
-	return nil
+type chainInfo struct {
+	localChainIds   chainIDSet
+	localChainNames chainNameSet
+	chainIDMutex    sync.RWMutex
+	chainNameMutex  sync.RWMutex
 }
 
-type chainNameSet map[string]bool
+func (c *chainInfo) AddIfUnique(chainId uint64, chainName string) error {
+	// Check if chainId is unique
+	c.chainIDMutex.RLock()
+	if c.localChainIds[chainId] {
+		c.chainIDMutex.RUnlock()
+		return fmt.Errorf("Chain with ID %d is duplicated", chainId)
+	}
+	c.chainIDMutex.RUnlock()
 
-func (s chainNameSet) AddIfUnique(name string) error {
-	if s[name] {
-		return fmt.Errorf("Chain with name %s is duplicated", name)
+	// Check if chainName is unique
+	c.chainNameMutex.RLock()
+	if c.localChainIds[chainId] {
```

Comment:
Don't you want to check `c.localChainNames` here?

Also, where do you write to `c.localChainIds`?

Reply:
This was a copy-paste error. moved to `sync.Map` per your other suggestion, this is no longer relevant.